python/lsst/dbb/buffmngrs/endpoint/backfill/commands.py

```diff
@@ -0,0 +1,139 @@
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Command line interface for the Backfill component.
+"""
+import click
+import importlib
+import inspect
+import jsonschema
+import logging
+import yaml
+from sqlalchemy import create_engine, inspect
+from sqlalchemy.orm import sessionmaker
+from .backfill import Backfill
+from .. import validation
+from ..utils import (
+    dump_config,
+    dump_env,
+    setup_logging,
+    find_missing_tables,
+    fully_qualify_tables,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@click.group()
+def backfill():
+    """Populate the database with entries for historical files.
+    """
+    pass
+
+
+@backfill.command()
+@click.option("--dump/--no-dump", default=True,
+              help="Log runtime environment and configuration "
+                   "(ignored if severity is set to WARNING and higher).")
+@click.option("--validate/--no-validate", default=False,
+              help="Validate configuration before starting the service.")
+@click.argument("filename", type=click.Path(exists=True))
+def start(filename, dump, validate):
+    """Starts a finder using a configuration from FILENAME.
+    """
+    with open(filename) as f:
+        configuration = yaml.safe_load(f)
+    if validate:
+        schema = yaml.safe_load(validation.backfill)
+        try:
+            jsonschema.validate(instance=configuration, schema=schema)
+        except jsonschema.ValidationError as ex:
+            msg = f"configuration error: {ex.message}."
+            logger.error(msg)
+            raise ValueError(msg)
+        except jsonschema.SchemaError as ex:
+            msg = f"schema error: {ex.message}."
+            logger.error(msg)
+            raise ValueError(msg)
+        return
+
+    config = configuration.get("logging", None)
+    setup_logging(options=config)
+
+    if dump:
+        msg = "runtime environment and configuration:\n\n"
+        msg += dump_env()
+        msg += "\n"
+        msg += dump_config(configuration)
+        logger.info(msg)
+
+    logger.info("setting up database connection...")
+    config = configuration["database"]
+
+    module = importlib.import_module("sqlalchemy.pool")
```

Comment:
Nothing in the _framework_ depends on LSST Stack software. Only ingest plugins use LSST code. I'd prefer to keep it this way in case plugins will end up in a separate module.